File: app/src/main/java/com/example/mike/mp3player/client/views/fragments/MediaControlsFragment.java
package com.example.mike.mp3player.client.views.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.mike.mp3player.R;
import com.example.mike.mp3player.client.activities.MediaActivityCompat;
import com.example.mike.mp3player.client.views.buttons.RepeatOneRepeatAllButton;
import com.example.mike.mp3player.client.views.buttons.ShuffleButton;
import com.example.mike.mp3player.dagger.components.MediaActivityCompatComponent;

import javax.inject.Inject;

public class MediaControlsFragment extends AsyncFragment {
    private static final String LOG_TAG = "PLY_PAUSE_BTN";
    private RepeatOneRepeatAllButton repeatOneRepeatAllButton;
    private ShuffleButton shuffleButton;


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        initialiseDependencies();
        super.onCreateView(inflater, container, savedInstanceState);
        return inflater.inflate(R.layout.fragment_media_controls, container, true);
    }

    @Override
    public void onViewCreated(View view, Bundle bundle) {
        super.onViewCreated(view, bundle);
        this.repeatOneRepeatAllButton.init(view.findViewById(R.id.repeatOneRepeatAllButton));
        this.shuffleButton.init(view.findViewById(R.id.shuffleButton));
    }


    public void initialiseDependencies() {
        MediaActivityCompatComponent component = ((MediaActivityCompat)getActivity())
                .getMediaActivityCompatComponent();
        component.playbackButtonsSubcomponent()
                .inject(this);
    }

    @Inject
    public void setRepeatOneRepeatAllButton(RepeatOneRepeatAllButton repeatOneRepeatAllButton) {
        this.repeatOneRepeatAllButton = repeatOneRepeatAllButton;
    }

    @Inject
    public void setShuffleButton(ShuffleButton shuffleButton) {
        this.shuffleButton = shuffleButton;
    }
}

File: client/src/main/java/com/github/goldy1992/mp3player/client/views/fragments/MediaControlsFragment.kt
package com.github.goldy1992.mp3player.client.views.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.github.goldy1992.mp3player.client.MediaControllerAdapter
import com.github.goldy1992.mp3player.client.databinding.FragmentMediaControlsBinding
import com.github.goldy1992.mp3player.client.views.buttons.RepeatOneRepeatAllButton
import com.github.goldy1992.mp3player.client.views.buttons.ShuffleButton
import com.github.goldy1992.mp3player.commons.LogTagger
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint(Fragment::class)
class MediaControlsFragment : Hilt_MediaControlsFragment(), LogTagger {

    @Inject
    lateinit var mediaControllerAdapter: MediaControllerAdapter

    @Inject
    lateinit var repeatOneRepeatAllButton: RepeatOneRepeatAllButton
    @Inject
    lateinit var shuffleButton: ShuffleButton

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val binding =  FragmentMediaControlsBinding.inflate(inflater)
        this.repeatOneRepeatAllButton.init(binding.repeatOneRepeatAllButton)
        this.shuffleButton.init(binding.shuffleButton)
        mediaControllerAdapter.repeatMode.observe(viewLifecycleOwner, repeatOneRepeatAllButton)
        mediaControllerAdapter.shuffleMode.observe(viewLifecycleOwner, shuffleButton)
        return binding.root
    }

    override fun logTag(): String {
        return "MDIA_CTRL_FGMT"
    }
}